### PR TITLE
[M3-8991 + M3-6911] - New Select Component + Managed Replacement

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
This PR introduces a new Select component aimed to facilitate the full deprecation of `react-select`.

This new component wraps Autocomplete to make a "simple" Select for the following reasons:
- Our new Select needs more capabilities than a simple select (easily disabling search, being clearable, being "creatable") in order to fulfill some of the UX requirements of `react-select`
- Wrapping `Autocomplete` allows us to 
  - only pick the options we are interested in
  - facilitates the developer experience for simple Select cases
  - Customize this instance only without adding further complexity to `Autocomplete` (for instance, adding the "creatable" logic to our existing `Autocomplete` would further complicate its maintenance)
  - Keep styling and common props consistency (helper text, error text, loading state, required state etc) without having to redefine them all over

In addition of adding a Story and relevant test, this PR replaces `react-select` in the Managed feature so we can see it in action in Cloud Manager.

> [!NOTE]
> This Select will also replace existing instances of `<Textfield select />`

## Changes  🔄

List any change(s) relevant to the reviewer.
- Add new Select component in `@linode/ui`
- Add new story
- Add unit test
- Add component test
- Replace instances in Managed

## Preview 📷
Screenshot 2024-12-12 at 11 41 52

### Verification steps
- Verify component behavior in Storybook: http://localhost:6006/?path=/docs/components-selects-select--documentation
- Verify component implementation and behavior in Managed:
  - /managed/monitors 👉 "Add Monitor" then "Edit Monitor" once created (**don't forget** to delete your bogus Monitor in order to avoid a failing one that's going to send warnings to the Managed team)
  - /managed/contacts 👉 "Add Contact" then "Edit Contact" once created

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

</details>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

### CODE REVIEW

**Review by harsh-akamai — APPROVED** [review]
Thanks for this @abailly-akamai! 🎉

File: packages/manager/src/features/Managed/MonitorDrawer.tsx [hunk 1]
```diff
@@ -157,7 +162,7 @@ const MonitorDrawer = (props: MonitorDrawerProps) => {
               />
 
               <Select
```
Comment (mjac0bs) [inline packages/manager/src/features/Managed/MonitorDrawer.tsx:164]: Should we make the Contact Group searchable to provide parity with prod?

| Prod | This Branch |
|--------|--------|
| <video src="https://github.com/user-attachments/assets/cc108cfe-dc3e-48d8-9557-e036f662b43f"/> | <video src="https://github.com/user-attachments/assets/97cf501f-e191-4155-a77c-980f6f93951a"/> |

**Review by mjac0bs — APPROVED** [review]
Thanks for the thorough tests and Storybook documentation. 🧪 

A few minor comments, but overall this looked good from my testing and read-through!

Note: Unrelated to this PR, I made M3-9021 - Fix Managed Contacts table in Mobile and Tablet views because they need some attention.

File: packages/manager/src/features/Managed/MonitorDrawer.tsx [hunk 2]
```diff
@@ -183,17 +187,16 @@ const MonitorDrawer = (props: MonitorDrawerProps) => {
...
                 <Grid sm={6} xs={12}>
                   <Select
-                    onChange={(item: Item<ServiceType>) =>
+                    onChange={(_, item: Item<ServiceType>) =>
                       setFieldValue('service_type', item.value)
                     }
                     textFieldProps={{
                       required: mode === modes.CREATING,
                     }}
+                    clearable={false}
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/features/Managed/MonitorDrawer.tsx]: sure, removed

File: packages/ui/src/components/Select/Select.test.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    expect(select).toHaveAttribute('readOnly');
+
+    expect(getByText('My Select')).toBeInTheDocument();
+    expect(getByRole('button', { name: 'Open' })).toBeInTheDocument();
+
+    const selectInput = getByRole('combobox');
+
+    options.forEach((option) => {
+      fireEvent.focus(selectInput);
+      fireEvent.change(selectInput, { target: { value: option.label } });
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/ui/src/components/Select/Select.test.tsx]: yesssss - done

File: packages/manager/src/features/Managed/MonitorDrawer.tsx (hunk 1 above)
Comment (abailly-akamai) [reply packages/manager/src/features/Managed/MonitorDrawer.tsx:164]: yup, done!

**Pull request merged**